src/ucs/sys/ucs_rtnetlink.c

```diff
@@ -0,0 +1,135 @@
+/**
+* Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2024. ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <ucs/debug/log.h>
+#include <ucs/debug/memtrack_int.h>
+#include <ucs/sys/ucs_rtnetlink.h>
+#include <ucs/sys/ucs_netlink.h>
+#include <ucs/sys/sock.h>
+#include <ucs/type/status.h>
+
+#include <errno.h>
+#include <linux/rtnetlink.h>
+
+#define NETLINK_MESSAGE_MAX_SIZE 8195
+
+
+struct ucs_netlink_route_info_t {
+    const struct sockaddr *sa_remote;
+    int                    if_index;
+    int                    matching;
+};
+
+static ucs_status_t ucs_rtnetlink_get_route_info(struct rtattr *rta, int len,
+                                                 int *if_idx, void **dst_in_addr)
+{
+    *if_idx      = -1;
+    *dst_in_addr = NULL;
+
+    for (; RTA_OK(rta, len); rta = RTA_NEXT(rta, len)) {
+        if (rta->rta_type == RTA_OIF) {
+            *if_idx = *((int *)RTA_DATA(rta));
+        } else if (rta->rta_type == RTA_DST) {
+            *dst_in_addr = RTA_DATA(rta);
+        }
+    }
+
+    if ((*if_idx == -1) || (*dst_in_addr == NULL)) {
+        ucs_diag("either iface index or dest address are missing "
+                  "in the routing table entry");
+        return UCS_ERR_INVALID_PARAM;
+    }
+
+    return UCS_OK;
+}
+
+static int ucs_rtnetlink_is_rule_matching(struct rtmsg *rtm, size_t rtm_len,
+                                          const struct sockaddr *sa_remote,
+                                          int iface_index)
+{
+    int   rule_iface;
+    void *dst_in_addr;
+
+    if (ucs_rtnetlink_get_route_info(RTM_RTA(rtm), rtm_len, &rule_iface,
+                                     &dst_in_addr) != UCS_OK) {
```

Comment:
minor: seems the code would be smaller if we removed ucs_rtnetlink_get_route_info helper function and put its code directly in ucs_rtnetlink_is_rule_matching, which is already small enough - but up to you